sotodlib/coords/det_match.py

```diff
@@ -0,0 +1,670 @@
+        if resfit_file is not None:
+            rs.add_resfit_data(resfit_file)
+        if bgmap_file is not None:
+            rs.add_bgmap_data(bgmap_file)
+        return rs
+
+    @classmethod
+    def from_solutions(cls, sol_file, north_is_highband=True, name=None, 
+                       fp_pars=None, platform='SAT'):
+        """
+        Creates an instance from an input-solution file. This will include both design data, along with smurf-band
+        and smurf-channel info. Resonance frequencies used here are the VNA
+        freqs measured by Kaiwen.
+
+        Args:
+            sol_file (str):
+                Path to solutions file
+            name (str):
+                Name to label this ResSet
+            north_is_highband (bool):
+                True if the north-side of the array corresponds to bands
+                4-8
+            fp_pars (dict):
+                Result of the function ``sotododlib.coords.optics.get_ufm_to_fp_pars``. If this is None, detector positions will
+                not be mapped to pointing angles.
+            platform (str):
+                'SAT' or 'LAT'. Used to determine which focal plane function to
+                use for pointing
+        """
+        resonances = []
+
+        if fp_pars is not None:
+            theta = -np.deg2rad(fp_pars['theta'])
+            dx, dy = fp_pars['dx'], fp_pars['dy']
+
+            if platform.upper() == 'SAT':
+                fp_func = optics.SAT_focal_plane
+            elif platform.upper() == 'LAT':
+                fp_func = optics.LAT_focal_plane
+    
+        with open(sol_file, 'r') as f:
+            labels  = f.readline().split(',') # Read header
+
+            i = 0
+            for line in f.readlines():
+                d = dict(zip(labels, line.split(',')))
+                if not d['bias_line']:  # Skip incomplete lines
+                    continue
+                
+                is_north = north_is_highband ^ (int(d['smurf_band']) < 4)
+                is_optical = (d['is_optical'].lower() == 'true')
+                r = Resonator(
+                    i, res_freq=float(d['freq_mhz']), smurf_band=int(d['smurf_band']),
+                    bg=int(d['bias_line']), det_x=float(d['det_x']),
+                    det_y=float(d['det_y']), det_rhomb=d['rhomb'],
+                    mux_band=int(d['mux_band']), mux_channel=int(d['mux_channel']),
+                    mux_layout_pos=int(d['mux_layout_position']),
+                    det_bandpass=d['bandpass'], det_pol=d['pol'],
+                    is_optical=is_optical,
+                    is_north=is_north
+                )
+
+                if int(d['smurf_channel']) != -1:
+                    r.smurf_channel = int(d['smurf_channel'])
+
+                if fp_pars is not None:
+                    x, y = float(d['det_x']), float(d['det_y'])
+                    xp = x * np.cos(theta) - y * np.sin(theta) + dx
+                    yp = x * np.sin(theta) + y * np.cos(theta) + dy
```

Comment:
There is a function in the `optics` module for this, might as well use it to keep duplicated code to a minimum

Reply:
Sorry I don't really see what function you're referring to... I see that `SAT_focal_plane` takes "mapping_data" that might be used for this, but not exactly sure how to use it or what its doing.

I think I'd rather just keep this here for now since its a simple transformation, but we can generalize later if we want.

Reply:
Was talking about [`ufm_to_fp`](https://github.com/simonsobs/sotodlib/blob/f82b2823e874bd6f0532375c5e4f4ec5eea2bc0b/sotodlib/coords/optics.py#L277). But its not a big deal. I have (unmerged) [a function](https://github.com/simonsobs/sotodlib/blob/c75cacd5ae4f487ac0c88c1fa3b4996636b35c30/sotodlib/coords/optics.py#L559) that provides a single interface for both the LAT and SAT that should eventually replace stuff like this, but that can wait until I finish that branch.